### PR TITLE
Report UPnP network gateways to stdout

### PR DESCRIPTION
**Pull request by tadaskay**
Fixes: #901

### CODE REVIEW

File: nat/upnp/discover.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	defer client.Close()
+	responses, err := ssdp.SSDPRawSearch(client, ssdp.UPNPRootDevice, 2, 3)
+	if err != nil {
+		return nil, err
+	}
+
+	var results []gatewayDevice
+
+	for _, response := range responses {
+		device := gatewayDevice{}
```
Comment (vkuznecovas) [inline nat/upnp/discover.go]: I'd extract the innards to something like this:
```golang
func SSDPResponseToDevice(response whateverTypeItActuallyIs) (device, error) {
                device := gatewayDevice{
                     server: response.Header.Get("server"),
                }

		loc, err := response.Location()
		if err != nil {
			return err
		}

		device.root, err = goupnp.DeviceByURL(loc)
		return device, err
}
```

then, the loop just becomes: 
```golang
for _, response := range responses {
       device, err := SSDPResponseToDevice(response)
       if err != nil {
           log ....
           continue
       }
        if !isGateway(device.root) {
			log.Debugf("%s not a gateway device: %v", logPrefix, device)
			continue
	}
	results = append(results, device)
}
```

File: cmd/di.go [hunk 2]
```diff
@@ -182,6 +183,8 @@ func (di *Dependencies) Bootstrap(nodeOptions node.Options) error {
 
 	di.PortPool = port.NewPool()
 
+	go upnp.ReportNetworkGateways()
```
Comment (vkuznecovas) [inline cmd/di.go]: Do we want to collect data from everyone  even though they are only consumers? Or is this only really useful when dealing with providers? Just asking if we should really call it here or not.

Comment (tadaskay) [reply cmd/di.go]: Thanks for pointing that out. I think it's fine to collect from consumers as well. Less logic branching, and anonymous data, so I don't see an issue here.

File: nat/upnp/discover.go (hunk 1 above)
Comment (tadaskay) [reply nat/upnp/discover.go]: Good suggestion, thanks

**Pull request merged**